### PR TITLE
Fix for item sorting in consume item inventory window

### PR DESCRIPTION
**Pull request by RDru**
#### Summary
```SUMMARY: Bugfixes "Fixed item sorting in 'Consume item' inventory window"```

#### Purpose of change
Item sorting in 'Consume item' inventory window was not working. It was semi-random. 

#### Describe the solution
Items are sorted by rotten first (you probably want to know if something is rotten), then items that will spoil, then items that could spoil but are preserved in container, then items that won't spoil. So basically items that you want to get rid or eat first are always at the front.
Each of that category is then sorted by spoil time.
And lastly items are sorted by name.

example

### CODE REVIEW

File: src/game_inventory.cpp [hunk 1]
```diff
@@ -555,33 +555,29 @@ class comestible_inventory_preset : public inventory_selector_preset
...
-            } else if( !container.type->container || !container.type->container->preserves ) {
-                if( it.is_fresh() ) {
-                    return 1;
-                } else if( it.is_going_bad() ) {
-                    return 3;
-                } else if( it.goes_bad() ) {
-                    return 2;
-                }
+        int get_order( const item_location &loc, const time_duration time ) const {
+            const item &it = const_cast<item &>( *loc );
```
Comment (BevapDin) [inline src/game_inventory.cpp]: You're casting `*loc` into a reference to non-const item, but assign it to a reference to const item. That makes no sense. The cast appears to be unnecessary.

Comment (RDru) [reply src/game_inventory.cpp]: yes, thanks for pointing out

File: src/game_inventory.cpp [hunk 2]
```diff
@@ -568,7 +568,7 @@ class comestible_inventory_preset : public inventory_selector_preset
 
     protected:
         int get_order( const item_location &loc, const time_duration time ) const {
-            const item &it = const_cast<item &>( *loc );
+            const item &it = static_cast<const item &>( *loc );
```
Comment (BevapDin) [inline src/game_inventory.cpp]: `*loc` already yields a `const item &`, why cast it into the very same thing?

Comment (RDru) [reply src/game_inventory.cpp]: I did not really checked what loc is

**Comment by kevingranade** [comment]
Since this is a menu for consuming, not dropping, shouldn't it prioritize things you want to eat rather than things you want to drop?
As it is if you have some reason to carry rotten items, they will always be pulled to the top of the menu and you have to skip over them to eat something.

**Comment by RDru** [comment]
I had few reasons:
- in most cases you want to get rid of them and this will inform you that you have them.  This is very useful if you choose to eat from many items, for example in your (mobile) base.
- it follows the logic of spoilage order - they are already spoiled so they are first. 
- I guess you want to eat rotten first when you can eat rotten (does rotten stuff turns into nothing after some time? I'm not sure of that)
- also in most cases you should have more not rotten than rotten items, so in that order you will see both on first page. In some other order you might not see rotten items at all in first page.

**Comment by kevingranade** [comment]
It's not the consume menu's job to tell you about questionably edible things, it's job is to present the things you are most likely to want to eat at the top.

**Comment by RDru** [comment]
So if I put them at the end would that be ok?

**Comment by kevingranade** [comment]
Rotten should go at the end in general.
Optionally, it could appear closer to the top if the player has a mutation that makes consuming rotten food beneficial.

**Comment by RDru** [comment]
All is made as suggested. Hoping for quick merge as I need this in my CDDA life :)

**Pull request merged**